torch/fx/passes/reinplace.py

```diff
@@ -0,0 +1,540 @@
+import torch
+from torch.fx import Node
+from torch.fx._compatibility import compatibility
+from torch._subclasses.fake_tensor import FakeTensorMode, FakeTensor
+from torch.utils._pytree import tree_map
+from torch.multiprocessing.reductions import StorageWeakRef
+
+
+
+import _operator
+from enum import Enum
+import itertools
+from typing import Optional, Set
+from collections import defaultdict
+
+__all__ = ['reinplace']
+
+class _ViewType(Enum):
+    NonView = 0
+    SingleOutputView = 1
+    MultiOutputView = 2
+
+def _is_view_op(tgt):
+    if tgt is not None and isinstance(tgt, torch._ops.OpOverload):
+        schema = tgt._schema
+        if len(schema.arguments) > 0:
+            first_arg = schema.arguments[0]
+            # check if op is a view
+            return first_arg.alias_info is not None and not first_arg.alias_info.is_write
```

Comment:
@anjali411 seems like a tag opportunity. In particular this test is a bit wonky because hypothetically someone could define a view where the aliased argument isn't the first one, and that would silently break the invariants here.